src/Components/NavBar/NavBar.jsx

```diff
@@ -125,20 +185,31 @@ export class NavBar extends Component {
  * @param {object} state
  * @returns {object} props
  */
-export const mapStateToProps = state => state;
+export const mapStateToProps = ({ navbar, profile, currentUser }) => ({
+  navbar,
+  profile,
+  currentUser,
+});
 
 /**
  * @param {function} dispatch
  * @returns {object} props
  */
 export const mapDispatchToProps = dispatch => ({
-  onToggleSideNav: isDrawerDisplay => dispatch(toggleSideNav(isDrawerDisplay)),
+  onToggleSideNav: () => dispatch(toggleSideNav()),
 });
 
 NavBar.propTypes = {
   onToggleSideNav: PropTypes.func.isRequired,
   navbar: PropTypes.object.isRequired,
   currentUser: PropTypes.object.isRequired,
+  profile: PropTypes.object.isRequired,
```

Comment:
does this mean we have to add profile prop when calling the Navbar component?

Reply:
Yes. @dusmel we agreed with @Karlmusingo, @giseleiradu that data from the user should come from one object called `profile` inside the initialState which looks like this:

```
const initialState = {
  currentUser: {
    isAuth: checkToken(),
    user: checkToken() || null,
    notification: {
      notificationList: [],
      notificationsCount: 0,
    },
  },
  navbar: {
    isDrawerDisplay: false,
  },
  profile: {
    user: {
      username: '@coolbeatz71',
      firstName: '',
      lastName: '',
      email: '',
      bio: undefined,
      image: undefined,
      followers: 0,
      following: 0,
      articles: 0,
      allowEmailNotification: false,
    },
  },
};
```